it/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnectorSpec.scala

```diff
@@ -105,8 +103,9 @@ class PersistenceConnectorSpec
       val body = Json.obj("messageId" -> messageId.value).toString()
 
       stub(OK, Some(body))
-
-      whenReady(connector.post(movementId, messageId, DeclarationAmendment, source).value) {
+      implicit val hc: HeaderCarrier =
+        HeaderCarrier(extraHeaders = Seq("X-Message-Type" -> MessageType.DeclarationAmendment.code, HeaderNames.CONTENT_TYPE -> MimeTypes.XML))
```

Comment:
You're partially defeating the object of the test here. Why are you adding these here when we should be testing that the connector is adding them?

We should probably also update the wiremock stub to check for appropriate headers and bodies -- after all, that's the whole point of this ticket.

Reply:
I agree and Done the changes as suggested